lib/util/memory.js

```diff
@@ -62,6 +62,17 @@ export function cleanMemory(makeNode) {
     }
   }
 
-  elementObject.cache.allocated.forEach(v => elementObject.cache.free.push(v));
+  // Empty all element allocations.
+  elementObject.cache.allocated.forEach(v => {
+    elementObject.cache.free.push(v);
+  });
+
   elementObject.cache.allocated.clear();
+
+  // Empty all attribute allocations.
+  attributeObject.cache.allocated.forEach(v => {
+    attributeObject.cache.free.push(v);
```

Comment:
Would be great to figure out a faster way of concatenating these Sets into the free array.